### ISSUE TITLE
Support for typed Tasks and type Client interface from SignalR Hub

### ISSUE THREAD

**Issue by smarek** [Dorin-Mocan/SignalRSwaggerGen#3]
Hi,

would it be possible to add schema documentation for typed Tasks responses and client interface ?

Consider following example

```csharp
    public interface IServicesClient
    {
        public Task<ConfirmMessage> ReceiveMessage(ChatMessage message);
    }

    [SignalRHub(path: "/websocket",documentNames: new []{"ws"})]
    public class MyWebSocket : Hub<IServicesClient>
    {

        [SignalRMethod(name:"subscribe", operationType:OperationType.Get)]
        public async Task<StatusMessage> Subscribe([SignalRArg] SubscribeMessage message)
        {
            await Groups.AddToGroupAsync(Context.ConnectionId, message.GroupName);
            return new StatusMessage(operationStatus: StatusMessage.CODE_SUCCESS);
        }
}
```

I'd like to see these in resulting OpenAPI document
  - GET method "subscribe" with schema of request (already works) and schema of response (not yet?)
  - Possibly different OpenAPI document describing the interface expected on Hub client (again with schemas of request/arguments and response)

Relevant docs
  - Strongly Typed Hubs https://docs.microsoft.com/en-us/aspnet/core/signalr/hubs?view=aspnetcore-5.0#strongly-typed-hubs
  - sample typed Hub Task response https://stackoverflow.com/a/36059732/492624

**Comment by Dorin-Mocan** [comment]
@smarek , I'll think about it and come back with an answer.

**Comment by Dorin-Mocan** [comment]
@smarek , I started implementing the required functionality, so stay tuned.

**Comment by smarek** [comment]
You're awesome, thank you very much

**Comment by Dorin-Mocan** [comment]
@smarek , just released version 2.1.2. Added support for method return type(SignalRReturnAttribute attribute). For the second point, you can decorate your interface with same attributes as your hub and it will work. Even though the package is called SignalRSwaggerGen, you can apply the attributes from this package to any types, not only classes inheriting from SignalR Hub class.

**Comment by smarek** [comment]
Love it, works out-of-the-box, I've already taken the approach of documenting client interface the same way as server hub
Since I do not have to manually add the response schemas to swagger registry, it's now very easy to use your great work!
